lib/bolt/target.rb

```diff
@@ -21,7 +21,34 @@ def self.parse_urls(urls)
     def initialize(uri, options = nil)
       @uri = uri
       @uri_obj = parse(uri)
+      @protocol = @uri_obj.scheme
+      @user = @uri_obj.user
+      @password = @uri_obj.password
+      @port = @uri_obj.port
       @options = options || {}
+      @configured = false
+    end
+
+    def update_conf(conf)
+      @protocol = conf[:transport] unless protocol
+
+      t_conf = conf[:transports][protocol.to_sym]
+      # Override url methods
+      url_keys = %i[user password port]
+      # not worth it?
+      # url_keys.each {|k| instance_variable_set(:"@#{k}", t_conf[k]) unless send(k)}
+      @user ||= t_conf[:user]
+      @password ||= t_conf[:password]
+      @port ||= t_conf[:port]
+
+      @options = t_conf.reject { |k, _| url_keys.include?(k) }.merge(@options)
+      @configured = true
+
+      self
+    end
+
+    def invalidate_conf
```

Comment:
Is this used for anything?

Reply:
no neither is @configured at this point. The idea was that "get_targets" would not search and update the targets configuration if it was already configured.